src/server/frontend_wayland/frame_executor.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2021 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: William Wold <william.wold@canonical.com>
+ */
+
+#ifndef MIR_FRONTEND_FRAME_CALLBACK_EXECUTOR_H
+#define MIR_FRONTEND_FRAME_CALLBACK_EXECUTOR_H
+
+#include <mir/executor.h>
+
+#include <memory>
+
+namespace mir
+{
+namespace time
+{
+class Alarm;
+class AlarmFactory;
+}
+
+namespace frontend
+{
+
+/// Runs frame callbacks that do not have a buffer to be attached to.
```

Comment:
I still don't like this comment. Documentation (and naming) should be about the feature an abstraction provides, not the initial usecase. In this case, the feature is delayed and throttled execution.